Assets/MixedRealityToolkit-SDK/Features/Boundary/System/MixedRealityBoundarySystem.cs

```diff
@@ -269,15 +280,30 @@ public int GetHashCode(object obj)
         /// </remarks>
         private const float boundaryObjectRenderOffset = 0.001f;
 
+        private GameObject boundaryVisualizationParent;
+
         /// <summary>
         /// Parent <see cref="GameObject"/> which will encapsulate all of the teleportable boundary visualizations.
         /// </summary>
-        private GameObject boundaryVisualizationParent;
+        private GameObject BoundaryVisualizationParent => boundaryVisualizationParent ?? (boundaryVisualizationParent = CreateBoundaryVisualizationParent());
+
+        private int ignoreRaycastLayerValue = -1;
 
         /// <summary>
         /// Layer used to tell the (non-floor) boundary objects to not accept raycasts
         /// </summary>
-        private readonly int ignoreRaycastLayer = LayerMask.NameToLayer("Ignore Raycast");
+        private int IgnoreRaycastLayerValue
```

Comment:
wondering if this should be a `LayerMask` instead of an `int`

Reply:
setting the layer on a GameObject needs the value, not the mask.

Reply:
understood. I needed to set the mask of the boundary system visualizations so that they would not interfere with the cursor / teleporting. at least it's a one time lookup.

Reply:
Do we want this to be configurable, like with focus?

![image](https://user-images.githubusercontent.com/13334553/44175288-9cbf7180-a09a-11e8-95ac-7d2200a35ad4.png)

Reply:
I don't believe so. The boundary visualizations should behave "transparently", meaning that they should not interfere with cursors or other objects in the scene (like the auto-boundary in the shell)

Reply:
Future proofing, in case the layer ever changes.

Reply:
I seriously doubt that, but then we could just change it, if it ever does change.

Reply:
This is a one time cost that ensures that we do not have to update.

Reply:
Defensive coding :)

Reply:
But at the cost of runtime perf.  Idk, I think in this case perf > once in a lifetime event

Reply:
how does this hurt perf when it is a "done once" task?

Reply:
we may be over-indexing a little on perf when we are discussing startup / one time lookups

Reply:
It has a cumulative impact, death by a thousand cuts.

Reply:
In this case, death by...one cut?

Reply:
I worry about how hard it would be to find these types of bugs if Unity opts to change but sure...